docs/source/libraries/keras.rst

```diff
@@ -10,64 +10,191 @@ and experimental neural network architectures.
 ------------------
 
-Currently ELI5 supports :func:`eli5.explain_prediction` for Keras image classifiers.
-:func:`eli5.explain_prediction` explains image classifications through `Grad-CAM <https://arxiv.org/pdf/1610.02391.pdf>`_.
+Currently ELI5 supports :func:`eli5.explain_prediction` for Keras image and text classifiers.
 
-The returned :class:`eli5.base.Explanation` instance contains some important objects:
+Explanations are done using the GradCAM_ technique. Using it, we feed an input into the network, and differentiate the model's output with respect to a hidden layer (that contains spatial information). We do a bunch of computations with those values and get a Grad-CAM "heatmap" (or "localization map"). The heatmap highlights what parts of the input (actually the parts of the hidden layer, but it can be resized) contributed to the prediction the most (positively or negatively).
 
-* ``image`` represents the image input into the model. A Pillow image.
+Roughly, the "formula" to calculate the localization map ``lmap`` is::
 
-* ``targets`` represents the explanation values for each target class (currently only 1 target is supported). A list of :class:`eli5.base.TargetExplanation` objects with the following attributes set:
+    lmap = relu(sum(w*A))
 
-    * ``heatmap``  a grayscale "localization map" (rank 2 (2D) numpy array, with float values in the interval `[0, 1]`). The numbers indicate how important the region in the image is for the target class (even if the target class was not the predicted class). Higher numbers mean that the region tends to increase the predicted value for a class. Lower numbers mean that the region has smaller effect on the predicted class score.
+where: 
 
-    * ``target`` the integer ID of the class (same as the argument to ``targets`` if one was passed, or the predicted class ID if no argument was passed).
+* ``w`` is the weights corresponding to activation maps ``A``.
+* ``A`` is the activation maps for the hidden layer, i.e. the output at that hidden layer for a given input.
+* ``relu`` is the ReLU_ rectifier operation (caps negative numbers at 0).
+    * For classification tasks, this can be thought as removing the influence on the explanation of other classes that may be present.
+* ``sum`` adds all its terms together into a single result.
+* ``w*A`` takes a linear combination of its terms.
 
-    * ``score`` the output of the network for the predicted class.
+To compute ``w``, we do::
+
+    w = pool(dy/dA)
+
+where:
+
+* ``dy/dA`` is the gradients of the output with respect to the activation maps.
+    * ``y`` may be a single target class in a classification task, i.e. a scalar value.
+* ``pool`` is the Global Average Pooling operation that takes gradients and averages them over certain axes.
+
+
+This is the formula presented in the Grad-CAM paper (https://arxiv.org/abs/1610.02391/).
+
+
+Depending on circumstances, in this library we may skip or modify some operations (like ReLU or pooling).
```

Comment:
Could we make "Depending on circumstances" more precise here, e.g. "Depending on options described below", if that's the case? Another interpretation is that this could depend on some heuristics or something else.

Reply:
Added more detail. Let me know if it can be rephrased.

Reply:
Looks great, thanks 👍